### PR TITLE
Update mosdepth

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

This PR updates mosdepth to version 0.3.3, and adds the functionality to generate d4 files.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Review by jemten — APPROVED** [review]
Reviewing after a request from @ramprasadn. I haven't used this program but the code looks good and shouldn't break compatibility with other pipelines. Maybe the original authors of this module have some input, @JoseEspinosa, @drpatelh?

The markdown lint action fails but this should be unrelated

File: modules/mosdepth/main.nf [hunk 1]
```diff
@@ -28,7 +29,13 @@ process MOSDEPTH {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def interval = window_size ? "--by ${window_size}" : "--by ${bed}"
+    if (window_size) {
+        interval = "--by ${window_size}"
+    } else if ( bed ) {
+        interval = "--by ${bed}"
+    } else {
+        interval = ""
```
Comment (drpatelh) [inline modules/mosdepth/main.nf:37]: I assume this means `mosdepth` will run without any intervals too? Can't remember now if that is possible. Maybe we should add a test for these different scenarios?

**Review by drpatelh — COMMENTED** [review]
Nice! Thanks @ramprasadn ! Just one comment :)

**Review by JoseEspinosa — COMMENTED** [review]
Thanks! LGTM once you address @drpatelh comment 🚀

File: modules/mosdepth/main.nf (hunk 1 above)
Comment (ramprasadn) [reply modules/mosdepth/main.nf:37]: Yes, it does. More tests are a good suggestion though, I have added couple more tests to test the following scenarios now. 

1. No intervals
2. Intervals from a bed file
3. Intervals from window

**Comment by drpatelh** [comment]
Thanks @ramprasadn ! And for fixing the markdownlint but maybe we should revert the last commit and fix the linting in another PR? Looks like that has opened a can of worms with the YAML lint?

**Comment by ramprasadn** [comment]
@drpatelh Yupe, absolutely! I couldn't merge the PR without fixing markdown lint, so I thought I should give a shot at fixing the error. But, well.. 😆

**Review by drpatelh — APPROVED** [review]
🚀

**Pull request merged**